### DISCUSSION TITLE
Excluded package patterns / zip file size too large

### DISCUSSION THREAD

**Discussion by sidewaysglance** [brefphp/bref#1172] [Support] [answered]
Hi, I run:
`composer require bref/bref bref/laravel-bridge --update-with-dependencies`

and then:
`php artisan vendor:publish --tag=serverless-config`

I then run `serverless deploy` and get:

`
Resource handler returned message: "Unzipped size must be smaller than 215465505 bytes (Service: Lambda, Status Code: 400, Request ID: xxx, Extended Request ID: null)" (RequestToken: xxx, HandlerErrorCode: InvalidRequest)
`

I can see that in the config file it shows:
```
package:
    # Directories to exclude from deployment
    patterns:
        - '!node_modules/**'
        - '!public/storage'
        - '!resources/assets/**'
        - '!storage/**'
        - '!vendor/**'
        - '!tests/**'
```

For some reason my zip file builds with `node_modules` excluded but `vendor` present. Excluding `vendor`, or even some some useless subfolders in various vendor packages would reduce the zip file size suitably.

Am I doing something wrong?

Thanks

**Comment by mnapoli** [comment, marked as answer]
This is weird, the `'!vendor/**'` line should exclude the vendor directory 🤔 

You can try debugging/experimenting by running `serverless package`. Then in the `.serverless/` directory you will find the zip file, you can check out its content without deploying.

I would try removing and re-adding lines in `package.patterns` one by one to see what gois wrong?

**Reply by sidewaysglance** [reply]
Thank you, I have managed get past this issue. I am not sure what I was doing wrong, but I changed the patterns to take out more specific vendor directories and it's doing as expected now. User error probably. Thanks
